modules/javafx.graphics/src/main/java/com/sun/javafx/application/PlatformImpl.java

```diff
@@ -1044,4 +1045,57 @@ private static boolean isSupportedImpl(ConditionalFeature feature) {
                 return Toolkit.getToolkit().isSupported(feature);
         }
     }
+
+    private static PlatformPreferences platformPreferences;
+
+    public static PlatformPreferences getPlatformPreferences() {
+        if (platformPreferences == null) {
+            throw new IllegalStateException("Toolkit not initialized");
+        }
+
+        return platformPreferences;
+    }
+
+    /**
+     * Called by Glass when the toolkit is initialized.
+     *
+     * @param wellKnownKeys a map of platform-specific keys to well-known keys
+     * @param preferences the initial set of platform preferences
+     */
+    public static void initPreferences(Map<String, String> wellKnownKeys, Map<String, Object> preferences) {
+        platformPreferences = new PlatformPreferences(wellKnownKeys);
+        platformPreferences.update(preferences);
+    }
+
+    /**
+     * Called by Glass when one or several platform preferences have changed.
+     * <p>
+     * This method can be called on any thread. The supplied {@code preferences} map may
+     * include all platform preferences, or only the changed preferences. If a preference
+     * was removed, the corresponding key is mapped to {@code null}.
+     *
+     * @param preferences a map that includes the changed preferences
+     */
+    public static void updatePreferences(Map<String, Object> preferences) {
+        if (isFxApplicationThread()) {
+            checkHighContrastThemeChanged(preferences);
```

Comment:
is this needed in this preferences-only PR?

Reply:
Yes. This PR replaces the special-purpose implementation to retrieve the theme name of the Windows platform with a general-purpse implementation to retrieve all kinds of preferences. We need to retain the theme name handling, otherwise high-contrast support for the Caspian and Modena themes doesn't work.